### ISSUE TITLE
Managing Dependencies

### ISSUE THREAD

**Issue by timothyas** [NOAA-PSL/graph-ufs#73]
TL;DR I think we should make WeatherBench2 an optional dependency, and I think there could be other packages that should be as well (I did this for torch after abandoning the torch dataloader).

The context for this is that I'm packaging up our code for other people to test it, and so trying to address #4 and make it generally easier to run this code. WeatherBench2 has 2 specific dependencies that I don't think we want:
1. It lists `jax[cpu]` as a dependency, which means that installing WeatherBench2 will conflict with our GPU version of jax
2. It lists a very specific version of xarray, which we may not want. For instance Zarr Python v3 is coming out soon, and I bet there are forthcoming changes to xarray that we will want.

Right now we have weatherbench2 as a submodule, so I guess it's not too bad to install it without its dependencies. However, it would be even smoother to be able to not require importing it for all workflows.
